packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -360,6 +365,111 @@ const QueryEditorExistingQueryHeader = observer(
   },
 );
 
+const QueryEditorExistingQueryInfoModal = observer(
+  (props: { existingEditorStore: ExistingQueryEditorStore; query: Query }) => {
+    const { existingEditorStore, query } = props;
+    const updateState = existingEditorStore.updateState;
+    const applicationStore = existingEditorStore.applicationStore;
+    const [queryVersionId, setQueryVersionId] = useState<string>(
+      existingEditorStore.query?.versionId ?? '',
+    );
+    const changeQueryVersionId: React.ChangeEventHandler<HTMLInputElement> = (
+      event,
+    ) => {
+      setQueryVersionId(event.target.value);
+    };
+
+    const updateQueryVersionId = applicationStore.guardUnhandledError(
+      async (): Promise<void> => {
+        flowResult(
+          existingEditorStore.updateState.updateQuery(
+            undefined,
+            queryVersionId,
+          ),
+        ).catch(applicationStore.alertUnhandledError);
+      },
+    );
+
+    const closeModal = (): void => updateState.setShowQueryInfo(false);
+
+    return (
+      <Dialog
+        open={updateState.showQueryInfo}
+        onClose={closeModal}
+        classes={{ container: 'search-modal__container' }}
```

Comment:
Just for my own understanding, how do we decide the names/syntax used in our class names?

Reply:
I just reuse some shared modals in the codebase